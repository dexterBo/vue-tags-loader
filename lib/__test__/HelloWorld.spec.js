const fs = require('fs');
const mockContext = require('./mock-context');
const loader = require('../index');
const store = require('../store');

it('记录vue文件中的所有的标签', async () => {
  store.clear();

  const callbackTplSpy = jest.fn();
  const context = mockContext({
    callbackSpy: callbackTplSpy,
    resourcePath: 'path/to/HelloWorld.vue',
  });
  const code = fs.readFileSync('lib/__test__/HelloWorld.vue', {
    encoding: 'utf8',
  });

  loader.call(context, code);

  expect(store.get('path/to/HelloWorld.vue')).toMatchSnapshot();
});

it('支持忽略文件', async () => {
  store.clear();

  const callbackTplSpy = jest.fn();
  const context = mockContext({
    callbackSpy: callbackTplSpy,
    resourcePath: 'path/to/HelloWorld.vue',
    exclude: ['path/to/HelloWorld.vue'],
  });
  const code = fs.readFileSync('lib/__test__/HelloWorld.vue', {
    encoding: 'utf8',
  });

  loader.call(context, code);

  expect(store.get('path/to/HelloWorld.vue')).toMatchSnapshot();
});
