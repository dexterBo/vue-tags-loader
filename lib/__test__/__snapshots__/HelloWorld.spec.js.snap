// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`支持忽略文件 1`] = `null`;

exports[`记录vue文件中的所有的标签 1`] = `
Array [
  "style",
  "template",
  "div",
  "a-form",
  "a-form-item",
  "a-input",
  "a-select",
  "a-select-option",
  "a-button",
  "el-select",
  "el-option",
  "el-input-number",
  "a-config-provider",
  "a-icon",
  "a-space",
  "a-page-header",
  "a-descriptions",
  "a-descriptions-item",
  "a",
  "br",
  "a-tag",
  "a-input-number",
  "a-row",
  "a-statistic",
  "a-form-model",
  "a-form-model-item",
  "a-date-picker",
  "a-switch",
  "a-checkbox-group",
  "a-checkbox",
  "a-radio-group",
  "a-radio",
  "script",
]
`;
